src/pip/_internal/resolution/resolvelib/factory.py

```diff
@@ -1,4 +1,5 @@
 import logging
+from distutils.version import LooseVersion
```

Comment:
As noted by @pfmoore, we should be using `packaging` here.

You can import `from pip._vendor.packaging.version import parse_version` and use that.

Reply:
Actually we can just do

```python
sorted(set(cand.version for cand in acands))
```

because `InstallationCandidate.version` is already a `packaging` version object.

Reply:
Done in e24e40dd9. It's `packaging.version.parse`, but `parse` was too generic, so I imported as `parse_version` anyway.

Reply:
@uranusjr Good point. Done in 8755b3d6e.